packages/event-cache/src/EventCache.js

```diff
@@ -18,18 +18,28 @@ const limiter = new Bottleneck({ maxConcurrent: 25 })
 
 const getPastEvents = memoize(
   async function(instance, fromBlock, toBlock, batchSize = 10000) {
-    if (!instance.loadedCache && instance.ipfsEventCache) {
+    if (
+      instance.ipfsEventCache &&
+      !instance.loadedCache &&
+      (!instance.cacheMaxBlock ||
+        instance.latestIndexedBlock < instance.cacheMaxBlock)
+    ) {
       try {
         debug('Loading event cache from IPFS', instance.ipfsEventCache)
         const cachedEvents = flattenDeep(
           await Promise.all(
             instance.ipfsEventCache.map(hash => get(instance.ipfsServer, hash))
           )
         )
-        fromBlock = cachedEvents[cachedEvents.length - 1].blockNumber + 1
-        debug(`Last cached blockNumber: ${fromBlock}`)
+        const lastCached = cachedEvents[cachedEvents.length - 1].blockNumber + 1
         debug(`Loaded ${cachedEvents.length} events from IPFS cache`)
-        await instance.backend.addEvents(cachedEvents)
+        debug(`Last cached blockNumber: ${fromBlock}`)
+        debug(`Latest indexed block: ${instance.latestIndexedBlock}`)
+        if (Number(lastCached) - 1 > Number(instance.latestIndexedBlock)) {
```

Comment:
nit: would it make sense to cast lastCached value to Number when it gets assigned on line 34 and not do any casting here. Just being paranoid since on line #41 fromBlock gets assigned lastCached and it may cause unforeseen issues down the line if it is a string ?